### ISSUE TITLE
Rule proposal: `prefer-node-protocol`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1198]
Node.js has added `node:` support for builtin modules https://nodejs.org/api/esm.html#esm_node_imports, I think add it is more clear that module is builtin.

## Fail

```js
import fs from 'fs';
```

```js
export {promises as default} from 'fs'
```

```js
async function foo() {
	const fs = await import('fs');
	// ...
}
```

## Pass

```js
import fs from 'node:fs';
```

```js
export {promises as default} from 'node:fs'
```

```js
async function foo() {
	const fs = await import('node:fs');
	// ...
}
```

**Comment by sindresorhus** [comment]
Great idea!

**Comment by fisker** [comment]
@sindresorhus I was thinking maybe `is-builtin-module` should handle this https://github.com/sindresorhus/is-builtin-module/issues/5

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by karlhorky** [comment]
@fisker was there any reason that `require` calls were not included in this? Eg. that `require('fs')` still passes with this rule?

**Comment by fisker** [comment]
Because require with `node:` only works in Node.js v16.

**Comment by fisker** [comment]
But maybe we can add it behind an option?

**Comment by karlhorky** [comment]
Ah yeah, that makes sense! I guess [backporting may be coming soon](https://twitter.com/jasnell/status/1385239259838222336), but in the meantime, keeping it behind an option would be cool!

**Comment by fisker** [comment]
Would you like to work on it?

**Comment by karlhorky** [comment]
While I would love to do this, currently have some really tight deadlines for creating new curriculum for our students - so earliest would probably be in a month or two.

**Comment by fisker** [comment]
Don't worry, I can work on it.

**Comment by karlhorky** [comment]
Amazing, thanks!

**Comment by arpowers** [comment]
this is a good rule, nice work

**Comment by DaSchTour** [comment]
This rule just broke my build. I think this should not be fixed by default with the recommended configuration. This is a quite new feature, that is not widely know and that may break the build if the environment is not prepared to handle node protocol imports.
